### PR TITLE
23354 - Misc updates for Notice of Withdrawal

### PR DESCRIPTION
**Pull request by chenhongjing**
*Issue #:* /bcgov/entity#23354

*Description of changes:*
- add filing type
- add fee codes
- add filing status
- add tests

Let me know if I'm missing something

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: legal-api/tests/unit/resources/v1/test_business_filings/test_filings.py [hunk 1]
```diff
@@ -1096,6 +1099,14 @@ def _fee_code_asserts(business, filing_json: dict, multiple_fee_codes, expected_
         ('BC1234567', CONTINUATION_OUT_FILING, 'continuationOut', Business.LegalTypes.BCOMP.value, False, []),
         ('BC1234567', CONTINUATION_OUT_FILING, 'continuationOut', Business.LegalTypes.BC_ULC_COMPANY.value, False, []),
         ('BC1234567', CONTINUATION_OUT_FILING, 'continuationOut', Business.LegalTypes.BC_CCC.value, False, []),
+        ('BC1234567', NOTICE_OF_WITHDRAWAL, 'noticeOfWithdrawal', Business.LegalTypes.COMP.value, False, []),
+        ('BC1234567', NOTICE_OF_WITHDRAWAL, 'noticeOfWithdrawal', Business.LegalTypes.BCOMP.value, False, []),
+        ('BC1234567', NOTICE_OF_WITHDRAWAL, 'noticeOfWithdrawal', Business.LegalTypes.BC_ULC_COMPANY.value, False, []),
+        ('BC1234567', NOTICE_OF_WITHDRAWAL, 'noticeOfWithdrawal', Business.LegalTypes.BC_CCC.value, False, []),
+        ('BC1234567', NOTICE_OF_WITHDRAWAL, 'noticeOfWithdrawal', Business.LegalTypes.CONTINUE_IN.value, False, []),
```
Comment (eason-pan-bc) [inline legal-api/tests/unit/resources/v1/test_business_filings/test_filings.py:1106]: the identifiers for these Continued In versions  starts with 'C', like 'C1234567'

File: legal-api/tests/unit/resources/v2/test_business_filings/test_filings.py [hunk 2]
```diff
@@ -1185,6 +1188,14 @@ def _get_expected_fee_code(free, filing_name, filing_json: dict, legal_type):
          False, [], False),
         ('CP1234567', CP_SPECIAL_RESOLUTION_TEMPLATE, 'specialResolution', Business.LegalTypes.COOP.value,
          False, ['SPRLN', 'OTCON'], False),
+        ('BC1234567', NOTICE_OF_WITHDRAWAL, 'noticeOfWithdrawal', Business.LegalTypes.COMP.value, False, [], False),
+        ('BC1234567', NOTICE_OF_WITHDRAWAL, 'noticeOfWithdrawal', Business.LegalTypes.BCOMP.value, False, [], False),
+        ('BC1234567', NOTICE_OF_WITHDRAWAL, 'noticeOfWithdrawal', Business.LegalTypes.BC_ULC_COMPANY.value, False, [], False),
+        ('BC1234567', NOTICE_OF_WITHDRAWAL, 'noticeOfWithdrawal', Business.LegalTypes.BC_CCC.value, False, [], False),
+        ('BC1234567', NOTICE_OF_WITHDRAWAL, 'noticeOfWithdrawal', Business.LegalTypes.CONTINUE_IN.value, False, [], False),
```
Comment (eason-pan-bc) [inline legal-api/tests/unit/resources/v2/test_business_filings/test_filings.py:1195]: same here

**Comment by chenhongjing** [comment]
Does anyone know what happened to CI linting(3.8)?

**Review by eason-pan-bc — APPROVED** [review]
LGTM (the linting issues found in CI are not in the new added code) 😃

**Comment by leodube-aot** [comment]
> Does anyone know what happened to CI linting(3.8)?

It looks like the [latest version of Pylint 3.x](https://pylint.readthedocs.io/en/latest/whatsnew/3/3.3/index.html) added the new `too-many-positional-arguments` check. You can see last week our CI started using that new version.

Screenshot 2024-09-26 at 9 23 08 AM

Screenshot 2024-09-26 at 9 24 13 AM

I think we should either ignore it globally by adding it to `disable` in `setup.fg`, or we can add individual disable comments.

**Comment by vysakh-menon-aot** [comment]
> > Does anyone know what happened to CI linting(3.8)?
> 
> It looks like the [latest version of Pylint 3.x](https://pylint.readthedocs.io/en/latest/whatsnew/3/3.3/index.html) added the new `too-many-positional-arguments` check. You can see last week our CI started using that new version.
> 
> Screenshot 2024-09-26 at 9 23 08 AM Screenshot 2024-09-26 at 9 24 13 AM
> I think we should either ignore it globally by adding it to `disable` in `setup.fg`, or we can add individual disable comments.

The best option would be create a separate ticket to fix these.

**Review by JazzarKarim — APPROVED** [review]

**Review by AimeeGao — APPROVED** [review]
LGTM (^_^)

**Review by leodube-aot — APPROVED** [review]

**Comment by chenhongjing** [comment]
> > > Does anyone know what happened to CI linting(3.8)?
> > 
> > 
> > It looks like the [latest version of Pylint 3.x](https://pylint.readthedocs.io/en/latest/whatsnew/3/3.3/index.html) added the new `too-many-positional-arguments` check. You can see last week our CI started using that new version.
> > Screenshot 2024-09-26 at 9 23 08 AM Screenshot 2024-09-26 at 9 24 13 AM
> > I think we should either ignore it globally by adding it to `disable` in `setup.fg`, or we can add individual disable comments.
> 
> The best option would be create a separate ticket to fix these.

Will take a look at it. Thanks, Leo and Vyaskh!

**Pull request merged**